virttest/utils_libvirt/libvirt_network.py

```diff
@@ -346,10 +346,29 @@ def change_tcp_config(expected_dict):
 
 def check_listen_address(params):
     """
-    Check listen address by ss command
+    Check listen address and port by ss command
 
     :param params: dictionary with the test parameter, get listen address
     """
-    listen_address = params.get("listen_address")
-
-    process.run(f"ss -tnap|grep qemu-kvm| grep '{listen_address}'", shell=True, ignore_status=False)
+    def _check_str_in_result(string, result):
+        """
+        Check string in result
+
+        :param string: the string to check
+        :param result: check string in result
+        """
+        check_list = eval(string)
+        for value in check_list:
+            ret = re.search(value, result)
+            if not ret:
+                raise exceptions.TestFail("Pattern '%s' is not matched in '%s'" % (value, result))
+
+    check_listening_on = params.get("check_listening_on")
```

Comment:
It seems to me that there are no essential difference between check_listening_on and check_established_conn. And you might need to check other keywords in future. They are just strings you want to check in ss command output. So I recommend we do not define so many paramaters. Instead, we define a common parameter. And the function name should be changed accordingly.

Reply:
+1

Reply:
Updated, thanks.